ts_src/psbt.ts

```diff
@@ -1746,49 +1736,110 @@ function randomBytes(options?: RngOpts): Buffer {
   return rng(32);
 }
 
-interface BlindingData {
-  value: string;
-  ag: Buffer;
-  abf: Buffer;
-  vbf: Buffer;
+export interface BlindingData {
+  satoshis: number;
+  amountBlinder: string;
+  asset: string;
+  assetBlinder: string;
 }
 
-function getBlindingDataForInput(
-  prevout: WitnessUtxo,
-  blindPrivKey?: Buffer,
+// Buffer = privateBlindingKey for conf inputs
+// BlindingData = blinders for already unblinded conf inputs
+// undefined = unconfidential inputs
+export type BlindingDataLike = Buffer | BlindingData | undefined;
+
+/**
+ * Compute outputs blinders
+ * @param inputsBlindingData the transaction inputs blinding data
+ * @param outputsData data = [satoshis, asset] of output to blind
+ * @returns an array of BlindingData[] corresponding of blinders to blind outputs specified in outputsData
+ */
+export function computeOutputsBlindingData(
+  inputsBlindingData: BlindingData[],
+  outputsData: Array<[number, string]>,
+): BlindingData[] {
+  const outputsBlindingData: BlindingData[] = [];
+  outputsData.forEach(([satoshis, asset], outputIndex) => {
+    const blindingData: BlindingData = {
+      satoshis,
+      asset,
+      assetBlinder: randomBlinder(),
+      amountBlinder: '', // invalid blinder at this step
+    };
```

Comment:
I would do something slightly different to increase the readability, like:
```suggestion
  outputsData.slice(0, outputsData.length -1).forEach(([satoshis, asset], outputIndex) => {
    const blindingData: BlindingData = {
      satoshis,
      asset,
      assetBlinder: randomBlinder(),
      amountBlinder: randomBlinder(),
    };
```

This way we can move the computation of the finalVBF out of the loop.